src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/IncrementalBuildTest.cs

```diff
@@ -1273,12 +1273,12 @@ public void ChangeSupportedAbis ()
 			var proj = new XamarinFormsAndroidApplicationProject ();
 			proj.SetAndroidSupportedAbis ("armeabi-v7a");
 			using (var b = CreateApkBuilder ()) {
-				b.Build (proj);
+				Assert.IsTrue (b.Build (proj), "First build should have succeeded.");
 
 				var parameters = Builder.UseDotNet ?
 					new [] { $"{KnownProperties.RuntimeIdentifier}=android.21-x86" } :
 					new [] { $"{KnownProperties.AndroidSupportedAbis}=x86" };
-				b.Build (proj, parameters: parameters, doNotCleanupOnUpdate: true);
+				Assert.IsTrue (b.Build (proj, parameters: parameters, doNotCleanupOnUpdate: true), "Second build should have succeeded.");
```

Comment:
One thing that is weird... `Build()` will always throw an exception unless you set `ThrowOnBuildFailure = false`.

I have no idea why we assert it returns `true` everywhere, because it will throw before we even assert? Maybe could just discard these changes for now?

Reply:
In some cases where we call build multiple times in a test we should be using the `Assert.IsTrue` method. Since if a build does fail we will have a good idea which one failed. The exception stack trace is not always enough when we allow it to throw. 

So tests with multiple "Build" calls should be using `ThrowOnBuildFailure = false;` I think, and messages should have things like "First build", "Second build" etc to make it easier to diagnose.

Reply:
At some point, it might be worth refactoring the tests to make them cleaner. 

Maybe an extension method on `ProjectBuilder` named `BuildAndAssert`:

```csharp
b.BuildAndAssert (proj, "first build");
//...
b.BuildAndAssert (proj, "second build");
```
The main issue is Xamarin.ProjectTools doesn't have a reference to NUnit.

Reply:
I backed these changes out to reduce the noise in the PR.